base/acme/src/main/java/org/dogtagpki/acme/server/ACMEEngineConfig.java

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright Red Hat, Inc.
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+//
+package org.dogtagpki.acme.server;
+
+import java.time.temporal.ChronoUnit;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author Endi S. Dewata
+ */
+@JsonInclude(Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown=true)
+public class ACMEEngineConfig {
+
+    private Boolean enabled = true;
+    private Boolean wildcard;
+
+    private Map<String, ACMEValidity> validities = new LinkedHashMap<>();
+
+    public Boolean isEnabled() {
+        return enabled;
+    }
+
+    public void setEnabled(Boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    public Boolean isWildcard() {
+        return wildcard;
+    }
+
+    public void setWildcard(Boolean wildcard) {
+        this.wildcard = wildcard;
+    }
+
+    public Map<String, ACMEValidity> getValidities() {
+        return validities;
+    }
```

Comment:
I disagree with using a map.  There are only 4 things that have expiration (nonces, authorizations, pending order and finalised order).  Each validity knob should be a field and furthermore they should always have a valid value, i.e. a sensible default that can *optionally* be overriden by configuration.

Reply:
OK, the map has been replaced with four individual validity fields with default values, and they can be overridden via engine.conf.

I've also moved the wildcard parameter into ACMEPolicyConfig inside ACMEEngineConfig per discussion with @cipherboy.

Reply:
Thank you @edewata!